apps/dolly-backend/src/main/java/no/nav/dolly/bestilling/fullmakt/FullmaktClient.java

```diff
@@ -0,0 +1,99 @@
+import no.nav.testnav.libs.data.pdlforvalter.v1.RelasjonType;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static org.apache.http.util.TextUtils.isBlank;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class FullmaktClient implements ClientRegister {
+
+    private final ErrorStatusDecoder errorStatusDecoder;
+    private final TransactionHelperService transactionHelperService;
+    private final FullmaktConsumer fullmaktConsumer;
+    private final PdlDataConsumer pdlDataConsumer;
+
+    @Override
+    public Flux<ClientFuture> gjenopprett(RsDollyUtvidetBestilling bestilling, DollyPerson dollyPerson, BestillingProgress progress, boolean isOpprettEndre) {
+
+        if (nonNull(bestilling.getFullmakt()) && !bestilling.getFullmakt().isEmpty()) {
+
+            return Flux.fromIterable(bestilling.getFullmakt())
+                    .flatMap(fullmakt -> {
+                        fullmakt.setFullmaktsgiver(dollyPerson.getIdent());
+                        if (isBlank(fullmakt.getFullmektig())) {
+                            return pdlDataConsumer.getPersoner(singletonList(dollyPerson.getIdent()))
+                                    .flatMap(person -> Flux.fromStream(person.getRelasjoner().stream()
+                                            .filter(relasjon -> relasjon.getRelasjonType().equals(RelasjonType.FULLMEKTIG))))
+                                    .next()
+                                    .map(relasjon -> {
+                                        fullmakt.setFullmektigsNavn(getFullName(relasjon.getRelatertPerson()));
+                                        fullmakt.setFullmektig(relasjon.getRelatertPerson().getIdent());
+                                        return fullmakt;
+                                    })
+                                    .switchIfEmpty(Mono.error(new DollyFunctionalException("Fant ikke fullmektig som relasjon for person " + dollyPerson.getIdent())));
+                        } else {
+                            return Mono.just(fullmakt);
+                        }
+                    })
+                    .collectList()
+                    .flatMapMany(fullmakter -> fullmaktConsumer.createFullmaktData(fullmakter, dollyPerson.getIdent()))
+                    .map(this::getStatus)
+                    .map(status -> futurePersist(progress, status));
+        }
+
+        return Flux.empty();
+    }
+
+    @Override
+    public void release(List<String> identer) {
+
+        identer.forEach(ident -> {
+            var fullmaktResponse = fullmaktConsumer.getFullmaktData(List.of(ident)).blockFirst();
+            fullmaktResponse.getFullmakt().forEach(fullmakt -> fullmaktConsumer.deleteFullmaktData(ident, fullmakt.getFullmaktId()).block());
```

Comment:
Bruk subscribe() i stedet for block(), og ikke bruk blockFirst()
subscribe() vil returnere med en gang og kjøre dette asynkront
block() gjør at dette vil henge og vente til ferdig og forsinke parallell kjøring med andre slettinger. 
Logg også gjerne slettingene.